### DISCUSSION TITLE
tutorials/federation-dgs/04-running-the-router

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#88] [course feedback]
# tutorials/federation-dgs/04-running-the-router

https://www.apollographql.com/tutorials/federation-dgs/04-running-the-router

**Comment by thejavamonk** [comment]
D:\learning\graphql\apollo-dgs\dgs-federation\router>router --config config.yaml
←[2m2024-02-11T05:18:46.969760Z←[0m ←[32mINFO←[0m ←[2m←[0m Apollo Router v1.36.0 // (c) Apollo Graph, Inc. // Licensed as ELv2 (https://go.apollo.dev/elv2)
←[2m2024-02-11T05:18:46.970138Z←[0m ←[32mINFO←[0m ←[2m←[0m Anonymous usage data is gathered to inform Apollo product development.  See https://go.apollo.dev/o/privacy for details.
←[2m2024-02-11T05:18:47.312809Z←[0m ←[31mERROR←[0m ←[2m←[0m uplink error, the request will not be retried: code=ACCESS_DENIED message=API key service:Music-Matcher-djgm1s:●●●●●●●●●●●●●●●●●●●VXQ  cannot access Uplink for the 'Music-Matcher-djgm1s' graph.
 Please verify that the graph exists and that the API key in question has Observer access or higher. code="APOLLO_ROUTER_LICENSE_INVALID"
←[2m2024-02-11T05:18:47.314754Z←[0m ←[32mINFO←[0m ←[2m←[0m stopped
←[2m2024-02-11T05:18:47.316900Z←[0m ←[31mERROR←[0m ←[2m←[0m no valid license was supplied
no valid license was supplied

I am trying to run this on windows i have set the values of APOLLO_KEY and APOLLO_GRAPH_REF using set command and I am able to see their values using echo %APOLLO_KEY% but getting the above issue any idea whats wrong ?

**Comment by rajatk16** [comment]
Quick FYI, make sure to check your router version using the command `./router --version`

For version 1.40.1, you might get an error saying `Segmentation Fault`

I installed an earlier version of router and it worked for me

``` curl -sSL https://router.apollo.dev/download/nix/v1.37.0 | sh```

**Comment by punjprakash** [comment]
https://community.apollographql.com/t/federation-with-java-dgs-getting-error-while-hitting-query-via-router/7535
